### ISSUE TITLE
Global vs Stratified Comparison

### ISSUE THREAD

**Issue by orhuna** [orhuna/ar_ivt#17]
Comparing stratified and global models.

Closing Criteria
- [x] Report optimal models for all 4 clusters and the global model
- [x] Bias-Variance Analysis of optimal models. Training and Test Accuracy wrt Sample Size [training %=0.1, 0.2, ..., 0.9]
- [x] Shapley and Variable Importance Comparison among all five models

**Comment by kellyrmat** [comment]
*Here's the Bias-Variance plots for the global and stratified XGB IVT models.*
*The testing-training splits vary from 10% to 90%*
*I calculated coefficient of determination by feeding the test/train data to the XGBmodel and recording each split's test/train validation score*

# Global XGB 
![image](https://github.com/orhuna/ar_ivt/assets/114206053/c5eab1d1-677f-4b24-b55c-25e6c8a8ad67)
---
# Stratified XGB

**Cluster 0:**
![image](https://github.com/orhuna/ar_ivt/assets/114206053/1abd2580-15c8-4d9c-b8f3-34b6e54deb81)

---
**Cluster 1:**
![image](https://github.com/orhuna/ar_ivt/assets/114206053/51c1978d-f415-461c-a587-cd623c88dbb7)

---
**Cluster 2:**
![image](https://github.com/orhuna/ar_ivt/assets/114206053/5599a5f1-23d4-44d3-bb45-ee2cf97544a4)

---
**Cluster 3:**
![image](https://github.com/orhuna/ar_ivt/assets/114206053/c790e6ee-9801-4d10-b2f7-7c1e553c6b41)

---

**Comment by kellyrmat** [comment]
# **Optimal Models:**
*I examine each XGBRegressor's parameters using the .get_params method*

## *Global Model*
XGBRegressor(base_score=None, booster=None, callbacks=None,
             colsample_bylevel=None, colsample_bynode=None,
             colsample_bytree=None, early_stopping_rounds=None,
             enable_categorical=False, eval_metric=None, feature_types=None,
             gamma=None, gpu_id=None, grow_policy=None, importance_type=None,
             interaction_constraints=None, learning_rate=0.387514395454483,
             max_bin=None, max_cat_threshold=None, max_cat_to_onehot=None,
             max_delta_step=None, max_depth=None, max_leaves=None,
             min_child_weight=None, missing=nan, monotone_constraints=None,
             n_estimators=432, n_jobs=None, num_leaves=20,
             num_parallel_tree=None, predictor=None, ...)
---
## *Cluster 0*
XGBRegressor(base_score=None, booster=None, callbacks=None,
             colsample_bylevel=None, colsample_bynode=None,
             colsample_bytree=None, early_stopping_rounds=None,
             enable_categorical=False, eval_metric=None, feature_types=None,
             gamma=None, gpu_id=None, grow_policy=None, importance_type=None,
             interaction_constraints=None, learning_rate=0.2283429858614914,
             max_bin=None, max_cat_threshold=None, max_cat_to_onehot=None,
             max_delta_step=None, max_depth=None, max_leaves=None,
             min_child_weight=None, missing=nan, monotone_constraints=None,
             n_estimators=81, n_jobs=None, num_leaves=20,
             num_parallel_tree=None, predictor=None, ...)
---
## *Cluster 1*
XGBRegressor(base_score=None, booster=None, callbacks=None,
             colsample_bylevel=None, colsample_bynode=None,
             colsample_bytree=None, early_stopping_rounds=None,
             enable_categorical=False, eval_metric=None, feature_types=None,
             gamma=None, gpu_id=None, grow_policy=None, importance_type=None,
             interaction_constraints=None, learning_rate=0.17253423655860106,
             max_bin=None, max_cat_threshold=None, max_cat_to_onehot=None,
             max_delta_step=None, max_depth=None, max_leaves=None,
             min_child_weight=None, missing=nan, monotone_constraints=None,
             n_estimators=384, n_jobs=None, num_leaves=27,
             num_parallel_tree=None, predictor=None, ...)
---
## *Cluster 2*
XGBRegressor(base_score=None, booster=None, callbacks=None,
             colsample_bylevel=None, colsample_bynode=None,
             colsample_bytree=None, early_stopping_rounds=None,
             enable_categorical=False, eval_metric=None, feature_types=None,
             gamma=None, gpu_id=None, grow_policy=None, importance_type=None,
             interaction_constraints=None, learning_rate=0.09414814607636392,
             max_bin=None, max_cat_threshold=None, max_cat_to_onehot=None,
             max_delta_step=None, max_depth=None, max_leaves=None,
             min_child_weight=None, missing=nan, monotone_constraints=None,
             n_estimators=321, n_jobs=None, num_leaves=31,
             num_parallel_tree=None, predictor=None, ...)
---
## *Cluster 3*
XGBRegressor(base_score=None, booster=None, callbacks=None,
             colsample_bylevel=None, colsample_bynode=None,
             colsample_bytree=None, early_stopping_rounds=None,
             enable_categorical=False, eval_metric=None, feature_types=None,
             gamma=None, gpu_id=None, grow_policy=None, importance_type=None,
             interaction_constraints=None, learning_rate=0.17253423655860106,
             max_bin=None, max_cat_threshold=None, max_cat_to_onehot=None,
             max_delta_step=None, max_depth=None, max_leaves=None,
             min_child_weight=None, missing=nan, monotone_constraints=None,
             n_estimators=384, n_jobs=None, num_leaves=27,
             num_parallel_tree=None, predictor=None, ...)
---

**Comment by kellyrmat** [comment]
@orhuna Just submitted PR containing this notebook

**Comment by orhuna** [comment]
@kellyrmat can you please add 95%/5% and 99%/1% to your training/test splits for the above?

**Comment by kellyrmat** [comment]
@orhuna yes, can do.

**Comment by kellyrmat** [comment]
@orhuna I just added the higher training rates. 


*Every cluster, except 2, had performance deficits at the 95/99% Split*

# Cluster 0
![image](https://github.com/orhuna/ar_ivt/assets/114206053/470374ae-9469-4d7d-914f-2ffa36fdd7bf)

---
# Cluster 1
![image](https://github.com/orhuna/ar_ivt/assets/114206053/a6c5ae51-eeee-4272-8083-671afaca34be)

---
# Cluster 2
![image](https://github.com/orhuna/ar_ivt/assets/114206053/37654ada-eb57-43ef-9932-12ea052f10c9)

---
# Cluster 3
![image](https://github.com/orhuna/ar_ivt/assets/114206053/08d150f4-065c-4c88-82ff-247cb11d9c92)

---
# Global 
![image](https://github.com/orhuna/ar_ivt/assets/114206053/509ef273-2808-4466-acd4-40af00f24cee)
---
